### PR TITLE
Add an article on the new TCP addon

### LINKED ISSUE #198: Add an article on the new TCP addon

### PR DESCRIPTION
**Pull request by johnsudaar**
Fixes #198

### CODE REVIEW

File: _posts/addons/2017-10-23-tcp-addon.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+through the dashboard on the addons section of your app under the `Network`
+category.
+
+### Setup configuration
+
+In order to use the Scalingo TCP addon, you will need to define a new container
+type. This can be done in your [`Procfile`](/internals/procfile.html)
+
+
+Your procfile should look like:
```
Comment (yannski) [inline _posts/addons/2017-10-23-tcp-addon.md]: Here is a sample Procfile for a NodeJS application:

File: _posts/addons/2017-10-23-tcp-addon.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
+---
+title: Scalingo TCP Addon
+modified_at: 2017-10-23 00:00:00
+category: addons
+tags: tcp load balancer addon custom
+---
+
+## Introduction
+
+By default, Scalingo will only let you run web application. Sometimes this is
```
Comment (yannski) [inline _posts/addons/2017-10-23-tcp-addon.md]: run web applications (meaning accessible through HTTP or HTTPS)

File: _posts/addons/2017-10-23-tcp-addon.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+Your procfile should look like:
+```
+web: npm start
+tcp: npm run start-tcp
+```
+
+### Contact your application
+
+When your application will start, we will define a `PORT` environment variable
```
Comment (yannski) [inline _posts/addons/2017-10-23-tcp-addon.md]: PORT (same name that you may be used to listen in your web containers)

File: _posts/addons/2017-10-23-tcp-addon.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+in your container. Your application must listen on this port in order to be
+accessible from the outside.
+
+Once up and running your application will be accessible at the URL defined by
+the `TCP_URL` environment variable. You can find this variable on our dashboard
```
Comment (yannski) [inline _posts/addons/2017-10-23-tcp-addon.md]: before how to find, provide a sample TCP_URL

File: _posts/addons/2017-10-23-tcp-addon.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+So if you have a addon with a limit of 1000 pps with 4 containers running, your
+bandwidth will be:
+
+`1000 * 1500 = 1,5 MB/s` for one container
+
+So `1,5 * 4 = 6 MB/s` in total
+
+## Samples
+
+You can find a sample NodeJS application that use this addon to expose a MQTT server built with Mosca [here](https://github.com/Scalingo/sample-node-mosca).
```
Comment (yannski) [inline _posts/addons/2017-10-23-tcp-addon.md]: Set the link text with keywords not "here"

File: _posts/addons/2017-10-23-tcp-addon.md (hunk 2 above)
Comment (johnsudaar) [reply _posts/addons/2017-10-23-tcp-addon.md]: Done

File: _posts/addons/2017-10-23-tcp-addon.md (hunk 1 above)
Comment (johnsudaar) [reply _posts/addons/2017-10-23-tcp-addon.md]: Done

File: _posts/addons/2017-10-23-tcp-addon.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+This `TCP_URL` looks like:
+`tcp://tcp-gateway-1.eu-fr-agora-1.scalingo.com:20003`
+
+You can find this variable on our dashboard or by using our CLI:
+
+```bash
+scalingo --app my-app env | grep TCP_URL
+```
+
+This URL is also accessible on the addon dashboard.
```
Comment (EtienneM) [inline _posts/addons/2017-10-23-tcp-addon.md]: This is kind of redundant with the above sentence `You can find this variable on our dashboard`. You can safely remove this one I think.

File: _posts/addons/2017-10-23-tcp-addon.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+  Your TCP containers might be restarted at any moment by our scheduler, this
+  is done with 0 downtime, but the connections to the old containers are
+  dropped. Hence your client must contain a retry policy in case a connection
+  is dropped.
+</blockquote>
+
+If you need to contact a `tcp` container from a `web` container, you must also
+use this URL, there is no direct link between your containers, every
+communication must pass via our load balancer. It's the exact same thing if you
+need to contact a`web` container from a `tcp` container, you must use the
```
Comment (EtienneM) [inline _posts/addons/2017-10-23-tcp-addon.md]: "a`web`" missing space

File: _posts/addons/2017-10-23-tcp-addon.md [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+If you have multiple `tcp` containers running, Scalingo distributes connections
+to those containers. Our load balancer distributes connections during the TCP
+handshake. When established, the communication is always routed to the same
+server.
+
+## Rate Limiting
+
+Your bandwidth to and from your containers is limited. This limit depends on
+your addon plan. This limit is expressed in packet per second per container. If
+a container gets over his limit, we will start dropping excess packet.
```
Comment (EtienneM) [inline _posts/addons/2017-10-23-tcp-addon.md]: packet -> packet**s**

File: _posts/addons/2017-10-23-tcp-addon.md [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+packet/s to byte/s by multiplying your rate limit by 1500.
+
+If you have an addon with a limit of 1000 packet/s with 4 containers running,
+your bandwidth is:
+
+`1000 * 1500 = 1.5 MB/s` for one container.
+
+`1.5 * 4 = 6 MB/s` for all your containers.
+
+A limit burst is granted for every container. This burst is equal to 10 time
```
Comment (EtienneM) [inline _posts/addons/2017-10-23-tcp-addon.md]: time -> times

File: _posts/addons/2017-10-23-tcp-addon.md [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+the limit, so if you have a limit of 10 packet/s, your burst will be set to 100
```
Comment (EtienneM) [inline _posts/addons/2017-10-23-tcp-addon.md]: packet -> packets

File: _posts/addons/2017-10-23-tcp-addon.md [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+packets.
+
+## Samples
+
+You can find a sample Node.js application that use this addon to expose a
```
Comment (EtienneM) [inline _posts/addons/2017-10-23-tcp-addon.md]: use -> uses

File: _posts/addons/2017-10-23-tcp-addon.md [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+title: Scalingo TCP Addon
+modified_at: 2017-10-23 00:00:00
+category: addons
+tags: tcp load balancer addon custom
+---
+
+## Introduction
+
+By default, Scalingo will only let you run application that communicate via
+HTTP or HTTPS. Sometimes this is not enough and you need to have application
```
[the remark this answers is not in the document]
Comment (johnsudaar) [reply _posts/addons/2017-10-23-tcp-addon.md]: Done

File: _posts/addons/2017-10-23-tcp-addon.md [hunk 13]
```diff
@@ -0,0 +1,103 @@
...
+that can accept raw TCP connections.
+
+This addon let you deploy application that can accept raw TCP connections.
```
[the remark this answers is not in the document]
Comment (johnsudaar) [reply _posts/addons/2017-10-23-tcp-addon.md]: Done

File: _posts/addons/2017-10-23-tcp-addon.md (hunk 6 above)
Comment (johnsudaar) [reply _posts/addons/2017-10-23-tcp-addon.md]: Nop, the first one is the Scalingo's dashboard and the second one is the Irma's dashboard

File: _posts/addons/2017-10-23-tcp-addon.md (hunk 7 above)
Comment (johnsudaar) [reply _posts/addons/2017-10-23-tcp-addon.md]: Done

File: _posts/addons/2017-10-23-tcp-addon.md (hunk 8 above)
Comment (johnsudaar) [reply _posts/addons/2017-10-23-tcp-addon.md]: Done

File: _posts/addons/2017-10-23-tcp-addon.md (hunk 9 above)
Comment (johnsudaar) [reply _posts/addons/2017-10-23-tcp-addon.md]: Done

**Comment by EtienneM** [comment]
LGTM

**Pull request merged**